geomstats/geometry/stratified/graph_space.py

```diff
@@ -381,141 +571,474 @@ def dist(self, graph_a, graph_b):
+            for base_graph_, graph_to_permute_ in zip(base_graph, graph_to_permute)
+        ]
+
+        self.perm_ = gs.array(perms[0]) if is_single else gs.array(perms)
+
+        return self._permute(metric, graph_to_permute, self.perm_)
+
+
+class _BasePointToGeodesicAligner(metaclass=ABCMeta):
+    """Base class for point to geodesic aligner.
+
+    Attributes
+    ----------
+    perm_ : array-like, shape=[...,n]
+        Node permutations where in position i we have the value j meaning
+        the node i should be permuted with node j.
+    """
+
+    def __init__(self):
+        self.perm_ = None
+
+    @abstractmethod
+    def align(self, geodesic, x):
+        raise NotImplementedError("Not implemented")
+
+    @abstractmethod
+    def dist(self, geodesic, x):
+        raise NotImplementedError("Not implemented")
+
+    def _get_n_points(self, x):
+        return 1 if gs.ndim(x) == 2 else gs.shape(x)[0]
+
+    def _permute(self, metric, graph_to_permute, perm):
+        return metric.space.permute(graph_to_permute, perm)
+
+
+class PointToGeodesicAligner(_BasePointToGeodesicAligner):
+    r"""Class for the Alignment of the points with respect to a geodesic.
+
+    Implementing the algorithm in [Huckemann2010] to select an optimal alignment to a
+    point with respect to a geodesic. The algorithm sample discrete set of n_points
+    along the geodesic between [s_min, s_max] and find the permutation that gets closer
+    to the datapoints along the geodesic.
+
+    Parameters
+    ----------
+    metric : GraphSpaceMetric
+    s_min : float.
+    s_max: float.
+    n_points : int.
+
+    References
+    ----------
+    .. [Calissano2020]  Calissano, A., Feragen, A., Vantini, S.
+        “Graph Space: Geodesic Principal Components for a Population of
+        Network-valued Data.” Mox report 14, 2020.
+        https://mox.polimi.it/reports-and-theses/publication-results/?id=855.
+    .. [Huckemann2010] Huckemann, S., Hotz, T., Munk, A.
+        "Intrinsic shape analysis: Geodesic PCA for Riemannian manifolds modulo
+        isometric Lie group actions." Statistica Sinica, 1-58, 2010.
+    """
+
+    def __init__(self, s_min, s_max, n_points=10):
+        super().__init__()
+        self.s_min = s_min
+        self.s_max = s_max
+        self.n_points = n_points
+        self._s = None
+
+    def __setattr__(self, attr_name, value):
+        r"""Set attributes."""
+        if attr_name in ["s_min", "s_max", "n_points"]:
+            self._s = None
+
+        return object.__setattr__(self, attr_name, value)
+
+    def _discretize_s(self):
+        r"""Compute the domain distretization."""
+        return gs.linspace(self.s_min, self.s_max, num=self.n_points)
+
+    @property
+    def s(self):
+        r"""Save the domain distretization."""
+        if self._s is None:
+            self._s = self._discretize_s()
+
+        return self._s
+
+    def _get_gamma_s(self, geodesic):
```

Comment:
gamma -> geodesic?

let's not use the name of the letters representing the concepts, but rather the concepts themselves